lib/api/utilities/text_renderer.rb

```diff
@@ -39,7 +39,11 @@ def initialize(text, format: nil, object: nil)
         @text = text
         @format = format
         @object = object
-        @project = object.project if object.respond_to?(:project)
+        if object.respond_to?(:project)
+          @project = object.project
+        else
+          @project = object
```

Comment:
Probably because you want to make sure that the `@project` is set if one is available...

Don't know what your research turned up when you looked down into the rabbit hole of text rendering